### DISCUSSION TITLE
Can you alter one of the pre designed .bin files?

### DISCUSSION THREAD

**Discussion by Anthony42022** [Jyers/Marlin#1972] [Q&A]
I am just starting on my 3D printing adventure and added a Swiss made Direct drive to ender 3 pro which is basically a V2 now.  I uploaded E3V2-BLTouch-5x5-v4.2.7-v2.0.1.bin firmware but the XY axis are not in the center of the bed.  I know how to change the Z offset from the display. I also figured I could probably just print something the hit the endstops at the correct points but would like to change the code.  I have seen the source code for marlin but am not sure if its completely built  like the E3V2-BLTouch-5x5-v4.2.7-v2.0.1.bin firmware is. I just want to alter the current firm but can not find the source code for  the E3V2-BLTouch-5x5-v4.2.7-v2.0.1.bin I already installed.  Does anyone have any suggestions.

**Comment by UliLac** [comment]
@Anthony42022 Easy, you can do it yourself. 
First, home your printer. 
Then Home Z, this will bring your nozzle to the "alleged center" of the bed according to the current settings.
Then, raise Z some, take a piece of paper and try to mark as exactky as possible the X and Y position the nozzle is on
Then, find the true bed center - basically, draw diagonals from top left to bottom right and top right to bottom left, where they cross the center is
Then measure the diversion, in mm, for X and Y from the position the nozzle is to the true center and enter the values measured in Control/Motion/Home Offset - from the nozzle to the left is a minus value, as is from the nozzle towards the front of the printer.

Trying to "draw" it ...

       ..........X             <-- Thats where your nozzle is (Home offset value would be 10 dots left, X = -10)
       .         
      O                     <-- this is where it needs to be (home offset would be two dots down, Y = -2)

A turorial and calibration .stl can be found here: https://www.thingiverse.com/thing:2280529

Cheers, Uli

**Reply by Anthony42022** [reply]
Dude.... I swear I tried that before and it didn't work. although  I don't think I stored the setting before.  I totally worked this time!  thank you for replying.

**Reply by UliLac** [reply]
👍
